---
title: "Processing gene expression data from switchgrass diversity panel"
author: "Colette Berg"
date: "2025-05-29"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Switchgrass G*E / eQTL analyses

# this code is for processing and normalizing gene expression data with DESeq, and then outputting it in a format compatible with matrixEQTL

```{r loading required packages}

library(edgeR)
library(dplyr)
library(reshape2)
library(ggplot2)
library(qqman)
library(DESeq2)

```

## read in the data and filter out indvs we won't use 

```{r read}
# read in and format the count data 
counts <- read.csv("data/switchgrass_counts_medFilt.txt", sep = "\t")

# exclude individuals that were vetted in the population structure exploration
# these indvs. didn't match their putative population, putative RNA clone in the paired habitat, putative DNA clone, or all three 
counts_filt <- dplyr::select(counts, -c("HYYUB", "NASNY", "HYYUC", "NASNZ",
                                        "HYCAT",
"NASPY",
"HYASN",
"HYAUP",
"HYBWZ",
"HYCSS",
"HZOYG",
"HYBAB",
"HYBXA",
"HYSXS",
"HYSBO",
"HYSUB",
"HYBOT",
"HYZHA",
"HYCSO",
"HYSPN",
"HYAUZ",
"HYAYB",
"HYYCX",
"NASUC",
"HYBWX",
"HYBNG",
"HYCWA",
"HYCBP",
"HYCXS",
"NABAP",
"NASTU"))
rownames(counts_filt) <- counts_filt$GeneID
counts_filt <- dplyr::select(counts_filt, -GeneID)

# read in and format the colData

coldata <- read.csv("data/sample_names_vetted.csv")
rownames(coldata) <- coldata$X
coldata <- dplyr::select(coldata, -X)

```


## make a dds object 

```{r dds}


# making a DESeq object
dds <- DESeqDataSetFromMatrix(countData = counts_filt,
                              colData = coldata,
                              design = ~ Site,
                              tidy=FALSE)


```


## normalize the RNA libraries

```{r normalize}
# variance stabilizing function to normalize
vsd <- vst(dds, blind = FALSE)


# make it a dataframe -- now going to split by habitat 
t_counts <- as.data.frame(t(as.data.frame(assay(vsd))))

```



## merge gene counts with ID and habitat; split data by habitat 


```{r initial merge}
# add back habitat and plant ID
merge <- cbind(coldata, t_counts)

# make one dataframe for Michigan reads
MI_normalized <- filter(merge, Site == "M")

# make one dataframe for Texas reads
TX_normalized <- filter(merge, Site == "P")


```


```{r}
fam_MI

famDat
```


## read in the .fam data from the bed files and number the rows 

```{r processing}

# process the fam file 
# this part ensures that gene counts are in the same order as in the vcf 

famDat <- read.csv("~/Dropbox/sample_fam.csv", header = FALSE)

colnames(famDat) <- c("SRR1", 
                     "SRR2", 
                     "plantID", 
                     "V4", 
                     "V5", 
                     "V6", 
                     "V7")

famDat <- mutate(famDat, num = row_number(famDat))

```

## merge fam files with gene counts (this takes a while)

```{r merge}

# merging fam file with the TX expression data, blank lines allowed (needs the exact same # of lines as the fam file.)

phen_fam_merged_TX <- merge(famDat, 
      TX_normalized, 
      all.x = TRUE,
      ID.vars = c("plantID"))

# merging fam file with the MI expression data, blank lines allowed (needs the exact same # of lines as the fam file.)

phen_fam_merged_MI <- merge(famDat, 
      MI_normalized, 
      all.x = TRUE,
      ID.vars = c("plantID"))
```



## re-order the merged files, take out unnecessary columns, and write to .csv

```{r ordering}

# order so it's the exact same order as the fam file 
order_TX <- phen_fam_merged_TX[order(phen_fam_merged_TX$num),]
order_MI <- phen_fam_merged_MI[order(phen_fam_merged_MI$num),]

# next step involves moving these to the cluster with fileZilla and changing the name to chrom-by-chrom (since each chrom needs a fam file)
# taking out the columns you don't need for matrix eQTL for MI & TX 

fam_MI <- dplyr::select(order_MI, 
                        -SRR1,
                        -V4,
                        -V5,
                        -V6,
                        -subpop,
                 -plantID,
                 -V7,
                 -Site,
                 -num)


fam_TX <- dplyr::select(order_TX, 
                        -SRR1,
                        -V4,
                        -V5,
                        -V6,
                        -subpop,
                 -plantID,
                 -V7,
                 -Site,
                 -num)

# writing the expression data as a matrix (for matrix eQTL)
write.csv(setNames(data.frame(t(fam_TX[,-1])), fam_TX[,1]),
          "data/genes_TX_matrix.csv")

write.csv(setNames(data.frame(t(fam_MI[,-1])), fam_MI[,1]),
        "data/genes_MI_matrix.csv")


```

```{r GxE_calcs}

# subtracting to get at GxE 

# prepping the MI dataframe
MI_p <- dplyr::select(fam_MI, -SRR2)
MI_mat <- as.matrix(MI_p)

# prepping the TX dataframe 
TX_p <- dplyr::select(fam_TX, -SRR2)
TX_mat <- as.matrix(TX_p)

# subtracting
GxE_mat <- MI_mat - TX_mat

# converting to dataframe
fam_GxE <- as.data.frame(GxE_mat)

# adding back in the SRA identity labels 
SRR <- as.data.frame(fam_TX$SRR2)
SRR$SRR2 <- SRR$`fam_TX$SRR2`
fam_GxE <- cbind(SRR, fam_GxE)
fam_GxE1 <- dplyr::select(fam_GxE, -`fam_TX$SRR2`)

# writing the GxE matrix for uploading to matrix eQTL
write.csv(setNames(data.frame(t(fam_GxE1[,-1])), fam_GxE1[,1]),
        "data/genes_GxE_matrix.csv")


```

```{r}

GxE <- setNames(data.frame(t(fam_GxE1[,-1])), fam_GxE1[,1])

GxE
```

```{r}

# the next step is to calculate the median coverage of each gene and plot it, spatially, across the chromosome 

# the goal is to compare gene expression in the N and K sub-genomes 

# with this finding I can follow up on the lovell et al. finding that there's more heritable climate-associated SNPs in the N subgenome than the K 


transposed_expression <- setNames(data.frame(t(fam_MI[,-1])), fam_MI[,1])
transposed_expression$median <- apply(transposed_expression, 1, median, na.rm=T)
transposed_expression$mean <- apply(transposed_expression, 1, mean, na.rm=T)

summarize <- dplyr::select(transposed_expression, median, mean)

# now to connect gene to bp position 

#genePos <- read.csv("data/AP13_gene_position.txt",
#                    sep="\t")

#expression_and_position_TX <- merge(summarize,
#      genePos,
#      all.x=TRUE,
#      ID.vars=c(geneid))

summarize$geneid <- rownames(summarize)


## adding a subgenome column 
KTX <- filter(expression_and_position_TX, 
       chr == "Chr01K" |
                        chr == "Chr02K" |
                        chr == "Chr03K" |
                        chr == "Chr04K" |
                        chr == "Chr05K" |
                        chr == "Chr06K" |
                        chr == "Chr07K" |
                        chr == "Chr08K" |
                        chr == "Chr09K")
KTX$subgenome <- "K"

NTX <- filter(expression_and_position_TX, 
       chr == "Chr01N" |
                        chr == "Chr02N" |
                        chr == "Chr03N" |
                        chr == "Chr04N" |
                        chr == "Chr05N" |
                        chr == "Chr06N" |
                        chr == "Chr07N" |
                        chr == "Chr08N" |
                        chr == "Chr09N")
NTX$subgenome <- "N"

# putting together the N and K subgenomes 
summTX <- rbind(KTX, NTX)

# plotting!
ggplot(summTX,
       aes(x=chr,
           y=median,
           fill = subgenome)) +
  theme_bw() +
  geom_boxplot() +
  scale_fill_viridis_d()


write.csv(summarize, "MI_median.csv")

```


## plot PCAs of gene expression 

```{r PCA}

# plot PCA of gene expression variance
plotPCA(vsd, intgroup=c("subpop", "Site"),
             returnData=FALSE) +
  scale_color_manual(values = c("grey",
                                "grey",
                                "brown",
                                "brown1",
                                "darkorange",
                                "darkorange3",
                                "dodgerblue2", 
                                "dodgerblue4",
                                "white",
                                "white")) +
  theme_bw()


```

## code graveyard 

```{r old code from vetting the data }
PCdat <- plotPCA(vsd, intgroup=c("subpop", "Site"),
             returnData=TRUE)


ggplot(PCdat,
       aes(x = PC1, y = PC2, col = Site)) + 
  geom_point() +
  scale_color_manual(values = c("purple",
                                "darkgreen")) +
  theme_bw() +
  ggtitle("Gene Expression PCA by Site")


ggplot(filter(PCdat, subpop=="Midwest"),
       aes(x = PC1, y = PC2,
           label = plantID,
           col = subpop)) + 
  geom_point() +
  theme_bw() +
  ggtitle("Gene Expression PCA by Site") + 
  geom_label()

ggplot(filter(PCdat, plantID=="J036.A" |
                                  plantID=="J396.A" |
                                  plantID=="J525.C" |
                                  plantID=="J526.C"),
       aes(x=PC1,
           y=PC2, col = subpop,
           label=plantID)) + 
  geom_point() +
  geom_label() +
  xlim(-25,50) +
  scale_color_manual(values=c("brown",
                              "dodgerblue")) +
  theme_bw()


## calc-ing euclidean distance 

checkDat <- read.csv("data/DESeq_PCA.csv")

ggplot(checkDat,
       aes(x=dist)) +
  geom_histogram() +
  theme_bw() +
  xlab("Euclidean distance between clones in MI/TX")
```

