---
title: "Scanning for cis & trans eQTLs in switchgrass diversity panel"
author: "Colette Berg"
date: "2025-06-26"
output: html_document
---

## load required packages and set working directory

```{r setup}
# loading in necessary packages 

knitr::opts_chunk$set(echo = TRUE)

library(MatrixEQTL)
library(dplyr)
library(ggplot2)
library(VariantAnnotation)
library(vcfR)
library(adegenet)
library(eQTLpipeline)
base.dir = find.package("MatrixEQTL")

# set the working directory
setwd("~/Dropbox/lowryLab")
```


## read in genotype data

```{r load genotype data, echo=FALSE, message=FALSE}

# setting genotype matrices 
# these were generated in awk on the MSU HPCC, from vcfs 
chrom1K = "~/Dropbox/lowryLab/data/Chr01K_Midwest_genotype_matrix.txt"

# loading genotype data 
allsnps = SlicedData$new();
allsnps$fileDelimiter = "\t";       # denote missing values;
allsnps$fileSkipColumns = 1;          # one column of row labels
allsnps$fileSliceSize = 10000;      # read file in pieces of 2,000 rows
allsnps$LoadFile(chrom1K)


```





## load the covariates file



## read in gene location and snp location file 

```{r snps_genes}

# gene name, chromosome, start, stop
gene_location_file_name = "data/AP13_gene_position.txt"

# snp name, chromosome, position (bp)
snps_location_file_name = "data/positions.txt"

# process smps
snpspos = read.table(snps_location_file_name, 
                     header = FALSE, 
                     stringsAsFactors = FALSE,
                     sep="\t")

# process gene names 
genepos = read.table(gene_location_file_name, 
                     header = TRUE, 
                     stringsAsFactors = FALSE,
                     sep="\t")

```

```{r}

# run the matrix eQTL function, scanning for cis and trans eQTLs 
me_MI_midwest = Matrix_eQTL_main(
  snps = allsnps,
  gene = geneMI_Midwest,
  cvrt = cvrt,
  snpspos = snpspos,
  genepos = genepos,
  cisDist = 1e6,
  output_file_name = "data/MI_Midwest_1K.txt",
  output_file_name.cis = "data/MI_Midwest_1K_cis.txt",
  pvOutputThreshold = pvOutputThreshold,
  pvOutputThreshold.cis = pvOutputThreshold,
  useModel = useModel,
  errorCovariance = errorCovariance,
  verbose = TRUE,
  pvalue.hist = "qqplot",
  min.pv.by.genesnp = FALSE,
  noFDRsaveMemory = TRUE)

```






```{r making maps of TX trans QTLs}

# read in the SNPs position as a dataframe 

snpsPos <- read.csv("data/positions.txt",
                    sep="\t",
                    header=FALSE)

# adding an extra column to SNPs pos for merging 
snpsPos$SNP <- snpsPos$V1

# read in the data output (one chrom at a time for now)
output1K <- read.csv("data/TX_1K.txt",
         sep="\t")

# I've already read in the gene position in another code block 
genePos$gene <- genePos$geneid

# merging eQTLs with gene position 
posMerge_1K <- merge(Chr01K_trans, genePos, ID.vars = c("gene"), all.x = TRUE)

# merging eQTLs with gene position and SNP position 
allMerge <- merge(posMerge_1K, snpsPos, ID.vars = SNP, all.x=TRUE)

# filtering out columns I don't need 
mergeFilt <- dplyr::select(allMerge, SNP, gene, p.value, chr, s1, V3)

# plotting! 1 chrom at a time for npw 
ggplot(filter(mergeFilt,
              chr != "scaffold_19",
              chr != "scaffold_34"),
       aes(x = V3/1000000,
           y = s1/1000000,
           color=-log10(p.value),
           alpha = -log10(p.value))) +
  geom_point() +
  theme_bw() +
  scale_color_viridis() +
  facet_wrap(~chr,
             nrow = 9) +
  ylab("Gene Position (Mb)") +
  xlab("Chr 01K Position (Mb)")
```




## run matrix eQTL (this will take many hours)

```{r run matrix eQTL for TX}

chrom4N = "~/Dropbox/lowryLab/data/Chr04N_genotype_matrix.txt"

# loading genotype data 
allsnps = SlicedData$new();
allsnps$fileDelimiter = "\t";       # denote missing values;
allsnps$fileSkipColumns = 1;          # one column of row labels
allsnps$fileSliceSize = 10000;      # read file in pieces of 2,000 rows
allsnps$LoadFile(chrom4N)



# run the matrix eQTL function, scanning for cis and trans eQTLs 
me_TX_4N = Matrix_eQTL_main(
  snps = allsnps,
  gene = geneTX,
  cvrt = cvrt,
  snpspos = snpspos,
  genepos = genepos,
  cisDist = 1e6,
  output_file_name = "data/TX_4N.txt",
  output_file_name.cis = "data/TX_4N_cis.txt",
  pvOutputThreshold = pvOutputThreshold,
  pvOutputThreshold.cis = pvOutputThreshold,
  useModel = useModel,
  errorCovariance = errorCovariance,
  verbose = TRUE,
  pvalue.hist = "qqplot",
  min.pv.by.genesnp = FALSE,
  noFDRsaveMemory = TRUE)


```



```{r}
dat <- read.csv("data/num_cis_trans_eQTLs.csv")

cis <- ggplot(dat, aes(x = as.factor(num), y = cis,
                col = subgenome,
                fill = subgenome)) + 
  geom_bar(stat="identity",
           position = "dodge") +
  theme_bw() +
  xlab("Chromosome Number") +
  ylab("# of cis eQTLs") +
  ggtitle("# of TX cis eQTLs") +
  scale_color_viridis_d() +
  scale_fill_viridis_d() 

trans <- ggplot(dat, aes(x = as.factor(num), y = trans,
                col = subgenome,
                fill = subgenome)) + 
  geom_bar(stat="identity",
           position = "dodge") +
  theme_bw() +
  xlab("Chromosome Number") +
  ylab("# of trans eQTLs") +
  ggtitle("# of TX trans eQTLs") +
  scale_color_viridis_d() +
  scale_fill_viridis_d() 


grid.arrange(cis, trans)

```



```{r run matrix eQTL for MI}

me_MI = Matrix_eQTL_engine(
  snps = snps,
  geneMI = geneTX,
  cvrt = cvrt,
  snpspos = snpspos,
  genepos = genepos,
  cisDist = 1e6,
  output_file_name = output_file_name,
  pvOutputThreshold = pvOutputThreshold,
  useModel = useModel,
  errorCovariance = errorCovariance,
  verbose = TRUE,
  pvalue.hist = TRUE,
  min.pv.by.genesnp = FALSE,
  noFDRsaveMemory = TRUE)

```


