---
title: "Processing gene expression data from switchgrass diversity panel"
author: "Colette Berg"
date: "2025-05-29"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# set working directory
root.dir = "~/Dropbox/LowryLab/data/"
```

## Switchgrass G*E / eQTL analyses

# this code is for processing and normalizing gene expression data with DESeq, and then outputting it in a format compatible with matrixEQTL

```{r loading required packages}

library(edgeR)
library(dplyr)
library(reshape2)
library(ggplot2)
library(qqman)
library(DESeq2)

```



## read in the data and only retain the vetted individuals

```{r read lowland}
# read in and format the count data 
counts <- read.csv("~/Dropbox/LowryLab/data/expression/switchgrass_counts_medFilt_lowland.csv")

# format
rownames(counts) <- counts$GeneID
counts <- dplyr::select(counts, -GeneID)

# read in and format the colData
coldata <- read.csv("~/Dropbox/LowryLab/data/RNA_sample_metadata_lowland.csv")
rownames(coldata) <- coldata$X
coldata <- dplyr::select(coldata, -X)

```


```{r}
dds
```


## make a dds object 

```{r dds}


# making a DESeq object - lowland
dds <- DESeqDataSetFromMatrix(countData = counts,
                              colData = coldata,
                              design = ~ Site,
                              tidy=FALSE)

```


## normalize the RNA libraries

```{r normalize}
# variance stabilizing function to normalize
vsd <- vst(dds, blind = FALSE)
# make it a dataframe -- now going to split by habitat 
t_counts <- as.data.frame(t(as.data.frame(assay(vsd))))

```



## merge gene counts with ID and habitat; split data by habitat 


```{r initial merge}
# add back habitat and plant ID
merge <- cbind(coldata, t_counts)

# make one dataframe for Michigan reads
MI_normalized <- filter(merge, Site == "M")

# make one dataframe for Texas reads
TX_normalized <- filter(merge, Site == "P")


```




## read in the .fam data from the bed files and number the rows 

```{r processing}

# process the fam file 
# this part ensures that gene counts are in the same order as in the vcf 

famDat <- read.csv("/Users/coletteberg/Dropbox/LowryLab/data/sample_fam_lowland.csv", header=FALSE)

colnames(famDat) <- c("SRR1", 
                     "SRR2", 
                     "plantID", 
                     "V4", 
                     "V5", 
                     "V6", 
                     "V7",
                     "pop",
                     "V8")

# adding num 
famDat$num <- row_number(famDat)

```

## merge fam files with gene counts (this takes a while)

```{r merge}

# merging fam file with the TX expression data, blank lines allowed (needs the exact same # of lines as the fam file.)

phen_fam_merged_TX <- merge(famDat, 
      TX_normalized, 
      all.x = TRUE,
      ID.vars = c("plantID"))

# merging fam file with the MI expression data, blank lines allowed (needs the exact same # of lines as the fam file.)

phen_fam_merged_MI <- merge(famDat, 
      MI_normalized, 
      all.x = TRUE,
      ID.vars = c("plantID"))
```





## re-order the merged files, take out unnecessary columns, and write to .csv

```{r ordering}

# order so it's the exact same order as the fam file 
order_TX <- phen_fam_merged_TX[order(phen_fam_merged_TX$num),]
order_MI <- phen_fam_merged_MI[order(phen_fam_merged_MI$num),]

# next step involves moving these to the cluster with fileZilla and changing the name to chrom-by-chrom (since each chrom needs a fam file)
# taking out the columns you don't need for matrix eQTL for MI & TX 

fam_MI <- dplyr::select(order_MI, 
                        -SRR1,
                        -SRR,
                        -V4,
                        -V5,
                        -V6,
                        -subpop,
                 -plantID,
                 -V7,
                 -V8,
                 -Site,
                 -num,
                 -pop)

fam_TX <- dplyr::select(order_TX, 
                        -SRR1,
                        -SRR,
                        -V4,
                        -V5,
                        -V6,
                        -subpop,
                 -plantID,
                 -V7,
                 -V8,
                 -Site,
                 -num,
                 -pop)

# writing the expression data as a matrix (for matrix eQTL)
write.csv(setNames(data.frame(t(fam_TX[,-1])), fam_TX[,1]),
          "~/Dropbox/LowryLab/data/expression/genes_TX_lowland.csv")

write.csv(setNames(data.frame(t(fam_MI[,-1])), fam_MI[,1]),
        "~/Dropbox/LowryLab/data/expression/genes_MI_lowland.csv")

```

```{r}
fam_TX
```


```{r GxE_calcs}

# subtracting to get at GxE 

# prepping the MI dataframe
MI_p <- dplyr::select(fam_MI, -SRR2)
MI_mat <- as.matrix(MI_p)

# prepping the TX dataframe 
TX_p <- dplyr::select(fam_TX, -SRR2)
TX_mat <- as.matrix(TX_p)

# subtracting
GxE_mat <- MI_mat - TX_mat

# converting to dataframe
fam_GxE <- as.data.frame(GxE_mat)

# adding back in the SRA identity labels 
SRR <- as.data.frame(fam_TX$SRR2)
SRR$SRR2 <- SRR$`fam_TX$SRR2`
fam_GxE <- cbind(SRR, fam_GxE)
fam_GxE1 <- dplyr::select(fam_GxE, -`fam_TX$SRR2`)

# writing the GxE matrix for uploading to matrix eQTL
write.csv(setNames(data.frame(t(fam_GxE1[,-1])), fam_GxE1[,1]),
        "~/Dropbox/LowryLab/data/expression/genes_GxE_lowland.csv")

```
